crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -2101,40 +2094,27 @@ impl AuthorityPerEpochStore {
         &self,
         keys: impl Iterator<Item = SequencedConsensusTransactionKey>,
     ) -> SuiResult<Vec<bool>> {
-        let size_hint = keys.size_hint().0;
-        let mut results = Vec::with_capacity(size_hint);
-        let mut fallback_keys = Vec::with_capacity(size_hint);
-        let mut fallback_indices = Vec::with_capacity(size_hint);
+        let keys = keys.collect::<Vec<_>>();
 
-        {
-            let consensus_quarantine = self.consensus_quarantine.read();
+        let consensus_quarantine = self.consensus_quarantine.read();
+        let tables = self.tables()?;
 
-            for (i, key) in keys.enumerate() {
-                if consensus_quarantine.is_consensus_message_processed(&key) {
-                    results.push(true);
+        Ok(do_fallback_lookup(
```

Comment:
just curious if `do_fallback_lookup` can be made to take an `impl Iterator` for the first arg instead of a slice, so that you can avoid the extra `collect`? seems like it should work; maybe it doesn't matter in practice, though.

Reply:
i tried doing this and quickly fell down a rustc rabbit hole - in the interests of time I think i'm going to leave this as is